src/connection-manager/agreement.ts

```diff
@@ -102,6 +104,82 @@ async function updateDailyAgreement(
   })
 }
 
+/**
+ * Detect the ledger before a flagged ledger, which will contain server version information.
+ *
+ * @param ledger_index - Index of current ledger.
+ * @returns Boolean.
+ */
+function preceedFlagLedger(ledger_index: string): boolean {
+  return parseInt(ledger_index, 10) % 256 === 255
+}
+
+/**
+ * Classify server version type.
+ *
+ * @param beta - Extracted 8-bit beta identifier.
+ * @throws InvalidVersionException.
+ * @returns String.
+ */
+function extractServerVersionType(beta: number): string {
+  if (beta >> 6 === 0) {
+    throw new Error('Unknown release type')
+  }
+
+  const isBeta = beta >> 6 === 1
+  const isRC = beta >> 6 === 2
+  const isRelease = beta >> 6 === 3
+
+  const prefix = isRelease ? '' : '-'
+  const releaseType = isBeta ? 'b' : isRC ? 'rc' : ''
+
+  const releaseNum = !isRelease ? beta & 0x3f : ''
+
+  const betaString = `${prefix}${releaseType}${releaseNum}`
+  return betaString
+}
+
+/**
+ * Decode server software version.
+ *
+ * @param server_version - Software server version in 64 bits integer.
+ * @throws InvalidVersionException.
+ * @returns String.
+ */
+function decodeServerVersion(server_version?: string): string {
+  if (!server_version) {
+    return ''
+  }
+
+  const num = BigInt(server_version)
+
+  const buf = Buffer.alloc(8)
+
+  buf.writeBigInt64BE(num)
+
+  if (buf.length !== 8) {
+    throw new Error('Size must be 8')
```

Comment:
I wouldn't throw, prefer to just return `null`

and then check for null as shown above.